pkg/sql/sem/tree/as_of.go

```diff
@@ -97,7 +97,11 @@ func EvalAsOfTimestamp(
 		}
 		// Attempt to parse as an interval.
 		if iv, err := ParseDInterval(s); err == nil {
```

Comment:
ParseDInterval is for parsing strings into postgres-compatible SQL datums, which happen to not support nanoseconds. We should consider changing this function to one that does support nanos, since that's a valid argument to AOST. We could just use go's time.ParseDuration for that, which would lose us support for days and months, but that probably doesn't matter since most GC intervals are less than that. Then again, why would anyone use AOST with a very specific nanosecond argument. Like, I can't think of a situation where that would actually matter. Ok so maybe this is fine? Just some thoughts.